### PR TITLE
Add scripts to build & publish our github actions, to 'bare tags' in the Khan/actions repo

### PR DESCRIPTION
**Pull request by jaredly**
## Summary:
Here's the magic! This will allow us to reference e.g. `Khan/actions@filter-files-v0.0.1` in our workflows (as described in the readme).
I've also added basic workflows, one to check for a changeset, and one to automate releases when a release PR lands.

Issue: https://khanacademy.atlassian.net/browse/FEI-4550

## Test plan:
See the workflow work! I published check-for-changeset-v0.0.0 manually from the CLI to demonstrate.

image

### CODE REVIEW

File: publish.mjs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+const checkTag = (tag) => {
+    try {
+        execSync(`git show-ref --tags ${tag}`);
+    } catch (err) {
+        return false;
+    }
+    return true;
+};
+
+const publishDirectoryAsTag = (distPath, tag, dryRun) => {
```
Comment (jeresig) [inline publish.mjs:22]: I think you meant to pass in `origin` here, as well.

File: build.mjs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        fs.rmSync(dist, { recursive: true });
+    }
+    copyDir(`actions/${name}`, dist);
+    if (packageJsons[name].dependencies) {
+        const replacements = [];
+        Object.keys(packageJsons[name].dependencies).forEach((depName) => {
+            replacements.push({
+                from: new RegExp(`\\buses: ${depName}\\b`, 'g'),
+                to: `uses: ${monorepoName}#${depName}-v${packageJsons[depName].version}`,
+            });
```
Comment (jeresig) [inline build.mjs:37]: Very interesting!

**Review by jeresig — APPROVED** [review]
Looking good to me!

A couple suggestions:
- We should probably have the scripts live in a directory - maybe "utils/"?
- Now that we have JS happening, we should probably have the usual eslint/prettier in place, at least.
- Even though the scripts are simple, it may be good to abstract out the logic into functions so that they're easier to test - we won't need to test the main scripts themselves, though.

File: publish.mjs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+const [_, __, ...args] = process.argv;
+const dryRun = args.includes('--dry-run');
+
+const packageNames = fs.readdirSync('actions');
+
+const packageJsons = {};
+packageNames.forEach((name) => {
+    const pkg = JSON.parse(
+        fs.readFileSync(`actions/${name}/package.json`, 'utf8'),
+    );
```
[the remark this answers is not in the document]
Comment (jaredly) [reply publish.mjs:47]: So requiring json doesn't work in esmodules mode 😬

[the remark this answers is not in the document]
Comment (jeresig) [reply publish.mjs:47]: lol! I'm not sure if there is much benefit to using esmodules mode - but I guess it's up to you!

File: build.mjs [hunk 4]
```diff
@@ -0,0 +1,47 @@
+/**
+ * Building a package currently only consists of processing
+ * the action.yml to handle any references to other actions in
+ * this repo.
+ */
+import fs from 'fs';
```
[the remark this answers is not in the document]
Comment (jaredly) [reply build.mjs:6]: yeah just modern import/exports. it's a nice lightweight to use ~modern js without needing to transpile.

[the remark this answers is not in the document]
Comment (jeresig) [reply build.mjs:6]: Ok!

**Pull request merged**